apitools/base/py/transfer.py

```diff
@@ -339,12 +341,18 @@ def __SetRangeHeader(self, request, start, end=None):
         else:
             request.headers['range'] = 'bytes=%d-%d' % (start, end)
 
-    def __GetChunk(self, start, end=None, additional_headers=None):
+    def __GetChunk(self, start, end=None, additional_headers=None,
+                   ignore_chunksize=False):
         """Retrieve a chunk, and return the full response."""
         self.EnsureInitialized()
         end_byte = end
-        if self.total_size and end:
-            end_byte = min(end, self.total_size)
+        if not ignore_chunksize:
```

Comment:
my thought was that i didn't want anyone to accidentally skip this logic (say if there were another method added).

i don't mind the naming bit -- it's figuring out *what* chunk to get. :grin: but i'm pulling it out to add tests.